docs/baseband/tutorials/getting_started.rst

```diff
@@ -0,0 +1,388 @@
+    ...                complex_data=fr.complex_data,
+    ...                bps=fr.bps, edv=fr.header0.edv,
+    ...                station=fr.header0.station, time=fr.time0)
+
+The minimal parameters needed to generate a file are listed under the
+documentation for each format's ``open``, though comprehensive lists can be
+found in the documentation for each format's stream writer class (eg. for
+VDIF, it's under `~baseband.vdif.base.VDIFStreamWriter`).  In practice we
+specify as many relevant header properties as available to obtain a particular
+file structure.  If we possess the *exact* first header of the file, it can
+simply be passed to ``open`` via the ``header`` keyword.  In the example above,
+though, we manually switch the values of ``nthread`` and ``nchan``.  Because
+VDIF EDV = 3 requires each frame's payload to contain 5000 bytes, and ``nchan``
+is now a factor of 8 larger, we decrease ``samples_per_frame``, the number of
+complete (i.e. all threads and channels included) samples per frame, by a
+factor of 8.
+
+Encoding samples and writing data to file is done by passing data arrays into
+``fw``'s `~baseband.vdif.base.VDIFStreamWriter.write` method.  The first
+dimension of the arrays is sample number, and the remaining dimensions must be
+as given by ``fw.sample_shape``::
+
+    >>> fw.sample_shape
+    SampleShape(nchan=8)
+
+In this case, the required dimensions are the same as the arrays from
+``fr.read``.  We can thus write the data to file using::
+
+    >>> while fr.tell() < fr.size:
+    ...     fw.write(fr.read(fr.samples_per_frame))
+    >>> fr.close()
+    >>> fw.close()
+
+For our sample file, we could simply have written
+
+    ``fw.write(fr.read())``
+
+instead of the loop, but for large files, reading and writing should be done in
+smaller chunks to minimize memory usage.  Baseband stores only the data frame
+or frame set being read or written to in memory.
+
+We can check the validity of our new file by re-opening it::
+
+    >>> fr = vdif.open(SAMPLE_VDIF, 'rs')
+    >>> fh = vdif.open('test_vdif.vdif', 'rs')
+    >>> fh.sample_shape
+    SampleShape(nchan=8)
+    >>> np.all(fr.read() == fh.read())
+    True
+    >>> fr.close()
+    >>> fh.close()
+
+File Format Conversion
+----------------------
+
+It is often preferable to convert data from one file format to another that
+offers wider compatibility, or better fits the structure of the data.  As an
+example, we convert the sample Mark 4 data to VDIF.
+
+Since we don't have a VDIF header handy, we pass the relevant Mark 4 header
+values into `vdif.open <baseband.vdif.open>` to create one. 
+
+    >>> import baseband.mark4 as mark4
+    >>> from baseband.data import SAMPLE_MARK4
+    >>> fr = mark4.open(SAMPLE_MARK4, 'rs', ntrack=64, decade=2010)
+    >>> spf = 640       # fanout * 160 = 640 invalid samples per Mark 4 frame
+    >>> f_rate = (fr.frames_per_second * fr.samples_per_frame / spf)*u.Hz
```

Comment:
Can we use `frames_per_second` instead? I'm still not sure which one to keep in the end, but it would seem to make sense to use the same argument in all examples....